lib/dynflow/persistence_adapters/sequel.rb

```diff
@@ -198,6 +198,16 @@ def push_envelope(envelope)
         table(:envelope).insert(prepare_record(:envelope, envelope))
       end
 
+      def prune_envelopes(receiver_ids)
+        connector_feature!
+        table(:envelope).where(receiver_id: receiver_ids).delete
+      end
+
+      def prune_undeliverable_envelopes
+        connector_feature!
+        table(:envelope).where(receiver_id: table(:coordinator_record).select(:id)).invert.delete
```

Comment:
This means we don't expect two coordinators running on top of the same database, right?
Could it be a problem in the future?

Reply:
Why would that be a problem?

Reply:
If we decide we want multiple coordinators for some reason.

Reply:
We already have multiple coordinators, in production deployments of Foreman you get a couple of them.

In order to be considered valid, it has to have a record in `dynflow_coordinator_records` table. This should affect only envelopes which don't have a receiver in that table. I'm still not following what the issue might be

Reply:
Oh, right, I've not read correctly the `invert`.